### PR TITLE
Add ios_user implementation module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

- ios_user implementation module
- integration test
- unit test

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

modules/network/ios/ios_user
##### ANSIBLE VERSION

```
devel 2.4
```

### CODE REVIEW

**Comment by gundalow** [comment]
```
lib/ansible/modules/network/ios/ios_user.py:168:9: E731 do not assign a lambda expression, use a def
lib/ansible/modules/network/ios/ios_user.py:169:9: E731 do not assign a lambda expression, use a def
lib/ansible/modules/network/ios/ios_user.py:170:9: E731 do not assign a lambda expression, use a def
lib/ansible/modules/network/ios/ios_user.py:244:4: E114 indentation is not a multiple of four (comment)
```

File: lib/ansible/modules/network/ios/ios_user.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    - username ansible secret password
+    - username admin secret admin
+"""
+
+import re
+
+from functools import partial
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ios import get_config, load_config
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ios/ios_user.py:146]: for declarative intent modules plan to use cliconf api's
Example usage:
```
from ansible.module_utils.connection import Connection
conn = Connection(module)
data = conn.get_config()
conn.edit_config(commands)
```

**Review by ganeshrn — CHANGES_REQUESTED** [review]
for declarative intent modules can use cliconf api's

**Review by rcarrillocruz — APPROVED** [review]

**Comment by nickrusso42518** [comment]
If possible, I am requesting an additional option to better secure password hashes.

```
ios_user:
  algorithm: ["md5", "sha256", "scrypt"]
```

These possible text values can be passed directly to the IOS CLI. The default (and weakest) option is MD5 which is what the module currently uses as the result of not specifying anything when the "secret" option is invoked. For those who work in high-security industries, maximum security for device passwords is a requirement.

The CLI syntax would be something like this:

`username admin2 privilege 15 algorithm-type scrypt secret MyPass123`

For module testing, "sha256" and "scrypt" use the digits 8 and 9, respectively, as seen below.

```
username admin1 secret 8 $8$krelHiyqebytyn$mzmwAnfueMwCiFQs96I89rSEENOvAQucSo99kxebfGs
username admin2 privilege 15 secret 9 $9$2VIfYlnbMhf6v9$/J6A7r7hgI7HvUB3Hsf5hiG0Rn7bn.VeY8VtfdRorBg
```

**Comment by trishnaguha** [comment]
@nickrusso42518 Acknowledged.

**Pull request merged**